tests/integration/test_stationboard.py

```diff
@@ -1,47 +1,112 @@
+
+        self.assertEqual(len(resp), 1)
+        self.assertEqual(resp[0]["id"], "ch:1:sloid:30813::1")
+        self.assertEqual(resp[0]["label"], "Hogwarts Express")
+        self.assertEqual(resp[0]["departureDate"], format_time(now))
+        self.assertEqual(resp[0]["estimatedDate"], format_time(now))
+        self.assertEqual(resp[0]["destinationName"], "Hogwarts")
+        self.assertEqual(resp[0]["destinationId"], "ch:1:sloid:91178::3")
+
+    @patch('chsdi.views.stationboard.get_current_registry')
+    def test_stationboard_nonexisting_id(self, mock_get_current_registry):
+        mock_get_current_registry.return_value = self.mock_registry
+        now = datetime.now(timezone('UTC')).strftime('%Y-%m-%dT%H:%M:%S.%f%z')
+        nonexisting_id = "99999"
+
+        # This will simulate the response of the OJP 2.0 API, when no station is found
+        mock_response = f"""<?xml version="1.0" ?>
+        <OJP xmlns:siri="http://www.siri.org.uk/siri" xmlns="http://www.vdv.de/ojp" version="2.0">
+            <OJPResponse>
+                <siri:ServiceDelivery>
+                    <siri:ResponseTimestamp>{now}</siri:ResponseTimestamp>
+                    <OJPStopEventDelivery>
+                        <siri:ResponseTimestamp>{now}</siri:ResponseTimestamp>
+                        <!-- No StopEventResult entries -->
+                    </OJPStopEventDelivery>
+                </siri:ServiceDelivery>
+            </OJPResponse>
+        </OJP>"""
+
+        with requests_mock.Mocker() as m:
+            m.post(self.mock_registry.settings['opentrans_url'], text=mock_response)
+            self.mock_request.matchdict = {"id": nonexisting_id}
+            view = TransportView(self.mock_request)
+
+            with self.assertRaises(HTTPNotFound) as e:
+                view.get_departures()
+            self.assertEqual(str(e.exception), f"No data available for the station {nonexisting_id}.")
+
+    @patch('chsdi.views.stationboard.get_current_registry')
+    def test_stationboard_wrong_id(self, mock_get_current_registry):
+        mock_get_current_registry.return_value = self.mock_registry
+        wrong_id = "lalala"
+        self.mock_request.matchdict = {"id": wrong_id}
+
+        with self.assertRaises(HTTPBadRequest) as e:
+            _ = TransportView(self.mock_request)
+        self.assertEqual(str(e.exception), "The id must be an integer.")
+
+    @patch('chsdi.views.stationboard.get_current_registry')
+    def test_stationboard_wrong_limit(self, mock_get_current_registry):
+        mock_get_current_registry.return_value = self.mock_registry
+        self.mock_request.params["limit"] = "lalala"
+        self.mock_request.matchdict = {"id": "123"}
+
+        with self.assertRaises(HTTPBadRequest) as e:
+            _ = TransportView(self.mock_request)
+        self.assertEqual(str(e.exception), "The limit parameter must be an integer.")
+
+    @patch('chsdi.views.stationboard.get_current_registry')
```

Comment:
This has been an original test.
Mocking it might be complicated, since if I understand the code correctly, the callback is added outside `TransportView()` on a more global level. Not exactly sure how to mock this. Any ideas @rebert @ltshb ?

Reply:
I think to keep it very simple the only thing you need to mock is the API call to the external service, no need to mock anything inside mf-chsdi3, simply mock the API at the request level. Of course you need to make sure that all API call are done via the `requests` library. So in the example above I don't understand the need to mock `chsdi.views.stationboard.get_current_registry` ?

Reply:
According to my sparse understanding of the old Pyramid code, the code of the tested classes needs values from the application's global registry (API key and URL). The registry itself is not available during the unit tests, hence I mocked it. But there might be more elegant ways to do that or probably ways to access the app's registry even when testing.

Reply:
I'll probably stick to the old `testapp` approach, as in the current tests for those tests